src/main/scala/com/fulcrumgenomics/sv/tools/AggregateSvPileup.scala

```diff
@@ -0,0 +1,319 @@
+    val pileupClusters: Map[BreakpointContigOrientation, Seq[PileupGroup]] =
+      pileupToNeighbors.view.mapValues(AggregateSvPileup.toClusters).toMap
+
+    // Aggregate each cluster
+    val aggregatedPileups: Map[BreakpointContigOrientation, Seq[AggregatedBreakpointPileup]] =
+      pileupClusters.view.mapValues(_.map(AggregatedBreakpointPileup.apply)).toMap
+
+    // Write output file
+    Metric.write[AggregatedBreakpointPileup](output, aggregatedPileups.values.flatten)
+  }
+
+}
+
+
+object AggregateSvPileup extends AggregateSvPileupDef {
+
+  /**
+   * Maps each pileup to the set of other pileups within the distance threshold
+   * @param pileups Set of pileups to evaluate for pairwise neighbor relationships
+   * @param maxDist Maximum distance (inclusive) for the left and right coordinates to be considered a neighbor
+   * @return Map of pileup to the set of neighbors
+   */
+  private def pileupToNeighbors(pileups: PileupGroup, maxDist: Int): Map[BreakpointPileup, PileupGroup] = {
+    val map: mutable.Map[BreakpointPileup, PileupGroup] = mutable.Map.from(pileups.map((_, Seq())))
+    for (pileup1 <- pileups; pileup2 <- pileups) {
+      if (pileup1.hashCode() < pileup2.hashCode() && isCluster(pileup1, pileup2, maxDist)) {
+        map.put(pileup1, map(pileup1).appended(pileup2))
+        map.put(pileup2, map(pileup2).appended(pileup1))
+      }
+    }
+    map.toMap
+  }
+
+  /** Returns true iff the two breakpoints can be clustered, i.e., they have the same left and right chromosomes and
+   * strand, and the distance between their left and right coordinates is below the threshold (inclusive) */
+  private def isCluster(bp1: BreakpointPileup, bp2: BreakpointPileup, maxDist: Int): Boolean = {
+    BreakpointContigOrientation(bp1) == BreakpointContigOrientation(bp2) &&
+      Math.abs(bp1.left_pos - bp2.left_pos) <= maxDist &&
+      Math.abs(bp1.right_pos - bp2.right_pos) <= maxDist
+  }
+
+  /**
+   * Converts a set of pairwise neighbor relationships to a set of clusters, where each cluster consists of breakpoints
+   * that can be joined into a connected path via pairwise neighbor relationships.
+   * @param pileupToNeighbors Map of pileup to its set of neighbors
+   * @return The set of pileup clusters
+   */
+  def toClusters(pileupToNeighbors: Map[BreakpointPileup, PileupGroup]): Seq[PileupGroup] = {
+    finalizeClusters(Seq(), pileupToNeighbors)
+  }
+
+  /**
+   * Similar to `toClusters()`, but picking up from a partially complete recursive call stack.
+   * @param existingClusters The set of clusters that have already been assembled
+   * @param remainingNeighborGroups The set of remaining neighbor groups, which are all separate from the existing
+   *                                clusters (don't belong with any of the existing clusters)
+   * @return The set of pileup clusters
+   */
+  @tailrec
+  private def finalizeClusters(existingClusters: Seq[PileupGroup],
```

Comment:
Why not just have these be one method with `existingClusters: Seq[PileupGroup] = Seq()`?